### PR TITLE
improvement(LogCollector): Added the log of the manager's scylla backend

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
Sometimes, the log of the manager's scylla backend is required for the investigations
of manager bugs, so I added its log to the monitor log set.

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by ShlomiBalalis** [comment]
Se the logs of 94adfe89-dcef-49d4-809c-66756be5e5c5 for the result:
https://cloudius-jenkins-test.s3.amazonaws.com/94adfe89-dcef-49d4-809c-66756be5e5c5/20230509_081646/monitor-set-94adfe89.tar.gz

File: sdcm/logcollector.py [hunk 1]
```diff
@@ -894,6 +894,9 @@ class MonitorLogCollector(LogCollector):
         FileLog(name='scylla_manager.log',
                 command='sudo journalctl -u scylla-manager.service --no-tail',
                 search_locally=True),
+        FileLog(name='manager_scylla_backend.log',
+                command='sudo journalctl -u scylla-server.service --no-tail',
```
Comment (fruch) [inline sdcm/logcollector.py:898]: Don't we need jmx and other parts ? I.e. same as we do for regular scylla

Comment (fgelcer) [reply sdcm/logcollector.py:898]: that is a good question, but after started, it does very little trouble, and the problem was during some tests, were the backend possibly had problems...
but yes, if not to hard, @ShlomiBalalis , please add all other backend logs, like we have for each db node

Comment (ShlomiBalalis) [reply sdcm/logcollector.py]: The only other service that exists in the monitor is scylla jmx. I added it as well.

**Review by fgelcer — APPROVED** [review]
LGTM

**Comment by fgelcer** [comment]
ping @fruch @roydahan , please review, merge and backport

**Review by roydahan — APPROVED** [review]

**Comment by fgelcer** [comment]
ping @scylladb/qa-maintainers to backport

**Pull request merged**